lib/core/src/regUtil.cpp

```diff
@@ -94,8 +94,16 @@ initCondForReg( rodsEnv *myRodsEnv, rodsArguments_t *rodsArgs, dataObjInp_t *dat
         addKeyVal( &dataObjOprInp->condInput, DATA_TYPE_KW, "generic" );
     }
 
-    if ( rodsArgs->collection == True ) {
-        addKeyVal( &dataObjOprInp->condInput, COLLECTION_KW, "" );
+    if (rodsArgs->collection == True) {
+        // The -r option has replaced -C, but we still support -C because we're nice.
+        rodsLogError(LOG_ERROR, DEPRECATED_PARAMETER, "-C option is deprecated. Use -r instead.\n");
```

Comment:
I realize that `rodsLogError` and `rodsLog` will be deprecated soon, but `addRErrorMsg` won't work here as this is before the API call wherein the structure is cleared and the message is lost. Open to other suggestions for what to use here.

Reply:
This seems fine.

Q. What does the output look like for this?
Q. Is `LOG_ERROR` the best choice for this message?

Reply:
Here's what the output looks like now...
```bash
$ ireg -C /var/lib/irods/regme /tempZone/home/rods/registered
remote addresses: 192.168.48.3 ERROR: -C option is deprecated. Use -r instead.
 status = -3000000 DEPRECATED_PARAMETER
```

I didn't give much thought to the '\n' til now. With it removed it looks like this (I think I like this better):
```bash
$ ireg -C /var/lib/irods/regme /tempZone/home/rods/registered
remote addresses: 192.168.48.3 ERROR: -C option is deprecated. Use -r instead. status = -3000000 DEPRECATED_PARAMETER
```

If I use `LOG_WARNING` or `LOG_NOTICE`, like I'd prefer, nothing is printed:
```bash
$ ireg -C /var/lib/irods/regme /tempZone/home/rods/registered
```

Every other use of `rodsLog` in this and the other `*Util.cpp` files uses `LOG_ERROR` so that the message is printed, so I was just following suit. But yes, I'd prefer if this was a `LOG_WARNING`- or `LOG_NOTICE`-level message.

In addition to this, the return code from using `-C` is 0. I think this is okay, but it might be unexpected if a user sees the error message.

Thoughts?

Reply:
I think a return code of zero is correct.

Given that 4.2.12 is likely the final release of the 4-2-stable branch and 4.3 is deprecating all of the rodsLog API, I think the output (without the newline) is acceptable.

Reply:
agreed, 0 seems correct.

WARNING seems ideal, but not sure how to get it there.

Reply:
We could always go with a `printf` or somesuch ;)

Reply:
I'm fine with a printf call. Maybe try comparing what you have to the deprecation message of `imeta adda` or some other icommand (i.e. `irm -n`?) that receives the message through the RsComm. I'd prefer that the formats look identical.

I also know that the format of the deprecation message is less important than actually informing the user that what they did is now deprecated. With that said, I'm fine with the current implementation.

We're going to replace the icommands with the new CLI anyway.

Reply:
lol Oh, boy... So let's say I start with this:
```bash
$ ils -Lr /tempZone/home/rods/registered_objects
/tempZone/home/rods/registered_objects:
  rods              0 demoResc            0 2022-02-23.19:23 & file_0
        generic    /var/lib/irods/regme/file_0
  rods              1 another            0 2022-02-23.19:24 & file_0
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_0
  rods              0 demoResc            0 2022-02-23.19:23 & file_1
        generic    /var/lib/irods/regme/file_1
  rods              1 another            0 2022-02-23.19:24 & file_1
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_1
  rods              0 demoResc            0 2022-02-23.19:23 & file_2
        generic    /var/lib/irods/regme/file_2
  rods              1 another            0 2022-02-23.19:24 & file_2
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_2
  rods              0 demoResc            0 2022-02-23.19:23 & file_3
        generic    /var/lib/irods/regme/file_3
  rods              1 another            0 2022-02-23.19:24 & file_3
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_3
  rods              0 demoResc            0 2022-02-23.19:23 & file_4
        generic    /var/lib/irods/regme/file_4
  rods              1 another            0 2022-02-23.19:24 & file_4
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_4
  rods              0 demoResc            0 2022-02-23.19:23 & file_5
        generic    /var/lib/irods/regme/file_5
  rods              1 another            0 2022-02-23.19:24 & file_5
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_5
  rods              0 demoResc            0 2022-02-23.19:23 & file_6
        generic    /var/lib/irods/regme/file_6
  rods              1 another            0 2022-02-23.19:24 & file_6
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_6
  rods              0 demoResc            0 2022-02-23.19:23 & file_7
        generic    /var/lib/irods/regme/file_7
  rods              1 another            0 2022-02-23.19:24 & file_7
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_7
  rods              0 demoResc            0 2022-02-23.19:23 & file_8
        generic    /var/lib/irods/regme/file_8
  rods              1 another            0 2022-02-23.19:24 & file_8
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_8
  rods              0 demoResc            0 2022-02-23.19:23 & file_9
        generic    /var/lib/irods/regme/file_9
  rods              1 another            0 2022-02-23.19:24 & file_9
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_9
```

`irm -U` prints a message and exits early with a code:
```bash
$ irm -U /tempZone/home/rods/registered_objects/file_9
-U is deprecated.  Please use iunreg instead.
$ echo $?
1
$ ils -L /tempZone/home/rods/registered_objects/file_9
  rods              0 demoResc            0 2022-02-23.19:23 & file_9
        generic    /var/lib/irods/regme/file_9
  rods              1 another            0 2022-02-23.19:24 & file_9
        generic    /var/lib/irods/anothervault/home/rods/registered_objects/file_9
```

`irm -n` prints a message via the rError stack and actually performs the action, return code of 0:
```bash
$ irm -n1 /tempZone/home/rods/registered_objects/file_9
Level 0: -n is deprecated.  Please use itrim instead.
$ echo $?
0
$ ils -L /tempZone/home/rods/registered_objects/file_9
  rods              0 demoResc            0 2022-02-23.19:23 & file_9
        generic    /var/lib/irods/regme/file_9
```

`imeta adda` performs the action, prints a message via rError with an "ERROR" label and has a return code of 0:
```bash
$ imeta adda -d /tempZone/home/rods/registered_objects/file_9 a v u
remote addresses: 192.168.80.3 ERROR: Level 0: "adda" is deprecated. Please use "add" with admin mode enabled instead.
$ imeta ls -d /tempZone/home/rods/registered_objects/file_9
AVUs defined for dataObj /tempZone/home/rods/registered_objects/file_9:
attribute: a
value: v
units: u
```

So... there's that.

Reply:
Hmmm... `irm -U` was dangerous - so we say 'nope'.

`irm -n` arguably should have done the same thing, but we didn't want to immediately break existing usage.

`imeta adda`, pretty sure nobody is using it - i'd be okay with it just doing the early exit thing like `irm -U`.

But yes, these are subtly different b/c severity and availability of a workaround.

Reply:
After some offline discussion, we have decided that this message will do:
```bash
$ ireg -C /var/lib/irods/regme /tempZone/home/rods/registered_objects
WARNING: -C option is deprecated. Use -r instead.
```